lib/index.js

```diff
@@ -267,11 +274,21 @@ export default {
             range = Helpers.generateRange(textEditor, line);
           }
 
+          // avoid duplicated issues
+          if (fingerprints.has(issue.fingerprint)) {
+            return;
+          }
+          fingerprints.add(issue.fingerprint);
+
           linterResults.push({
-            type: 'Warning',
-            text: issue.description,
-            filePath,
-            range,
+            severity: mapSeverity[issue.severity] || 'warning',
+            excerpt: `${issue.engine_name.toUpperCase()} [${issue.check_name}]: ${issue.description}`,
```

Comment:
In the last screenshot:

- `issue.engine_name.toUpperCase()` = `PHPMD`.
- `issue.check_name` = `Controversial/CamelCaseVariableName`.
- `issue.description` = `The variable $parentStyle is not named in camelCase`.

It allows me to know the engine that is throwing the issue (PHPMD), the ruleset and the rule that are throwing the issue (from the `check_name`) and an explanation of the issue (`description`).

It is true that after appending the whole CodeClimate detailed explanation in Markdown format, maybe it is not so crucial, because you can see it unfolding the extra information (the `message.description` is an accordion, it's not shown by default).

In the end I suppose it's just a matter of preference. Maybe another member of this package could provide us his/her POV :innocent: 

> just wondering if ... needs to be put behind a flag.

What do you mean? How would you do that?